tiledb/tests/test_webp.py

```diff
@@ -0,0 +1,134 @@
+from numpy.testing import assert_array_equal, assert_allclose
+import numpy as np
+import pytest
+import shutil
+import tempfile
+import tiledb
+
+
+@pytest.mark.parametrize(
+    "schema",
+    [
+        (np.int64, np.int64, None, True),   # Sparse arrays are not supported
+        (np.int64, np.int64, True, False),  # Variable sized attributes are not supported
+    ],
+)
+def test_webp_init(schema):
+    attr_dtype, dim_dtype, var, sparse = schema
+    with pytest.raises(tiledb.TileDBError):
+        tiledb.ArraySchema(
+            domain=tiledb.Domain(
+                [
+                    tiledb.Dim("y", domain=(1, 100), dtype=dim_dtype),
+                    tiledb.Dim("x", domain=(1, 300), dtype=dim_dtype),
+                ]
+            ),
+            attrs=[
+                tiledb.Attr("rgb", dtype=attr_dtype, var=var, filters=[tiledb.WebpFilter()])
+            ],
+            sparse=sparse
+        )
+
+
+def make_image_data(width, height, pixel_depth):
+    center_x = width / 2
+    center_y = height / 2
+
+    colors = {
+        "red": [255, 0, 0],
+        "green": [0, 255, 0],
+        "blue": [0, 0, 255],
+        "white": [255, 255, 255],
+        "black": [0, 0, 0]
+    }
+    if pixel_depth > 3:
+        for color in colors.values():
+            color.append(255)
+
+    rgb = []
+    for row in range(0, height):
+        r = []
+        for col in range(0, width):
+            if row < center_y and col < center_x:
+                r.append(colors["red"])
+            elif row < center_y and col > center_x:
+                r.append(colors["green"])
+            elif row > center_y and col < center_x:
+                r.append(colors["blue"])
+            elif row > center_y and col > center_x:
+                r.append(colors["white"])
+            elif row == center_y or col == center_x:
+                r.append(colors["black"])
+        rgb.append(r)
+    return rgb
+
+
+@pytest.mark.parametrize(
+    "image_dims",
+    [
+        (3, 7),
+        (20, 20),
+        (40, 40),
+        (479, 149),
+        (1213, 1357),
+        (1111, 3333),
+    ]
+)
+@pytest.mark.parametrize(
```

Comment:
Could this work instead?

```
@pytest.mark.parametrize(
    "colorspace", [
        tiledb.filter.lt.WebpInputFormat.WEBP_RGB,
        tiledb.filter.lt.WebpInputFormat.WEBP_BGR,
        tiledb.filter.lt.WebpInputFormat.WEBP_BGRA,
    ]
)
@pytest.mark.parametrize("lossless", [True, False])
```

Reply:
Sorry forgot to highlight all the relevant lines in my review😬

Reply:
No worries! Thanks for the review, I took care of a few comments and I'll rebase here shortly to clean up the rest. 

This is my first go with pybind and I'm not too sure how to conditionally skip these WebP tests.. In my most recent commit I did this, but this doesn't really work because it's possible the version is 2.13 or later and TileDB was built with `TILEDB_WEBP=OFF`. 

```
@pytest.mark.skipif(
    tiledb.libtiledb.version() < (2, 13, 0),
    reason="WebP filter supported in TileDB >= 2.13",
)
```

I see for serialization the tests are conditionally initialized using `TILEDB_SERIALIZATION` in `main.cc` and `setup.py` but I haven't had luck checking `TILEDB_WEBP` this way. Is there something I should do to expose the `TILEDB_WEBP` build option to pybind or is this usually passed to the TileDB-Py build command for serialization?

 `TILEDB_SERIALIZATION=ON python setup.py build --tiledb=/path/to/tiledb/dist/`

Reply:
Ah... yeah we should probably be checking if the TileDB passed was built with or without TILEDB_SERIALIZATION or TILEDB_WEBP.  If there is an easy way to expose TILEDB_WEBP in pybind11 that would be preferable, but if it seems really complicated then doing that latter but for webp is fine.